boxsdk/session/box_session.py

```diff
@@ -211,14 +173,38 @@ def _is_json_response(network_response):
         except ValueError:
             return False
 
-    def _retry_request_if_necessary(self, network_response, attempt_number, *args, **kwargs):
+    def _get_retry_after_time(self, attempt_number, retry_after_header):
         """
-        Retry a request for certain types of failure.
+        Get the amount of time to wait before retrying the API request.
+
+        If the Retry-After header is supplied, use it; otherwise, use exponential backoff
+        For 202 Accepted (thumbnail or file not ready) and 429 (too many requests), retry later, after a delay
+        specified by the Retry-After header.
+        For 5xx Server Error, retry later, after a delay; use exponential backoff to determine the delay.
+
+        :param attempt_number:          How many attempts at this request have already been tried.
+        :type attempt_number:           `int`
+        :param retry_after_header:      Value of the 'Retry-After` response header.
+        :type retry_after_header:       `unicode` or None
+        :return:                        Number of seconds to wait before retrying.
+        :rtype:                         `Number`
+        """
+        # pylint:disable=no-self-use
+        if retry_after_header is not None:
+            return float(retry_after_header)
+        return 2 ** attempt_number
+
+    def _get_retry_request_callable(self, network_response, attempt_number, request):
+        """
+        Get a callable that retries a request for certain types of failure.
+
         For 401 Unauthorized responses, renew the session by refreshing the access token; then retry.
         For 202 Accepted (thumbnail or file not ready) and 429 (too many requests), retry later, after a delay
         specified by the Retry-After header.
         For 5xx Server Error, retry later, after a delay; use exponential backoff to determine the delay.
 
+        Otherwise, return None.
```

Comment:
I might make this comment first. Most calls succeed. So mostly this method will return None.

Also makes me wonder if this should be called `def _get_retry_request_on_retriable_failures`
or `_get_retry_request_callable_on_retriable_failures` or `_get_retry_callable_on_retriable_failures`